src/kagglehub/gcs_upload.py

```diff
@@ -0,0 +1,154 @@
+    """upload files in a folder. Throws an exception if more than 50 files are found.
+    Parameters
+    ==========
+    folder: the folder to upload from
+    quiet: suppress verbose output (default is False)
+    model_type: Type of the model that is being uploaded.
+    """
+    # TODO(aminmohamed): Handle case where more than 50 files are to be uploaded by first zipping the content
+
+    # Count the total number of files
+    file_count = 0
+    for _, _, files in os.walk(folder):
+        file_count += len(files)
+
+    if file_count > MAX_NUM_FILES:
+        max_num_files_exception = "Cannot upload more than 50 files. Consider zipping the files first."
+        raise RuntimeError(max_num_files_exception)
+
+    tokens = []
+    for file_name in os.listdir(folder):
+        tokens.append(_upload_file_or_folder(folder, file_name, model_type, quiet))
+
+    return tokens
+
+
+def _upload_file_or_folder(
+    parent_path: str, file_or_folder_name: str, model_type: str, quiet: bool = False  # noqa: FBT002, FBT001
+):
+    """
+    Uploads a file or each file inside a folder individually from a specified path to a remote service.
+
+    Parameters
+    ==========
+    parent_path: The parent directory path from where the file or folder is to be uploaded.
+    file_or_folder_name: The name of the file or folder to be uploaded.
+    dir_mode: The mode to handle directories. Accepts 'zip', 'tar', or other values for skipping.
+    model_type: Type of the model that is being uploaded.
+    quiet: suppress verbose output (default is False)
+    :return: A token if the upload is successful, or None if the file is skipped or the upload fails.
+    """
+    full_path = os.path.join(parent_path, file_or_folder_name)
+    if os.path.isfile(full_path):
+        return _upload_file(file_or_folder_name, full_path, quiet, model_type)
+
+    elif os.path.isdir(full_path):
+        for filename in os.listdir(full_path):
+            file_path = os.path.join(full_path, filename)
+            if os.path.isfile(file_path):
+                _upload_file(filename, file_path, quiet, model_type)
+            elif not quiet:
+                logger.info(f"Skipping non-file item in directory: {filename}")
+    elif not quiet:
+        logger.info("Skipping: " + file_or_folder_name)
+    return None
+
+
+def _upload_file(file_name: str, full_path: str, quiet: bool, model_type: str):  # noqa: FBT001
+    """Helper function to upload a single file
+    Parameters
+    ==========
+    file_name: name of the file to upload
+    full_path: path to the file to upload
+    quiet: suppress verbose output
+    model_type: Type of the model that is being uploaded.
+    :return: None - upload unsuccessful; instance of UploadFile - upload successful
+    """
+
+    if not quiet:
+        logger.info("Starting upload for file " + file_name)
+
+    content_length = os.path.getsize(full_path)
+    token = _upload_blob(full_path, model_type)
```

Comment:
Looking at the implementation of `_upload_blob`, it will always return a token or raise. No need for `if token is None`.